src/freenet/node/Node.java

```diff
@@ -2539,6 +2533,46 @@ public void realRun() {
 		System.out.println("Node constructor completed");
 	}
 
+	// peers-offers/*.fref files
+	private void processPeersOffersFrefFiles(SubConfig nodeConfig, int configOptionSortOrder) {
+		nodeConfig.register("peersOffersDismissed", false, configOptionSortOrder, true, true,
+				"Node.peersOffersDismissed", "Node.peersOffersDismissedLong", new BooleanCallback() {
+
+					@Override
+					public Boolean get() {
+						synchronized(Node.this) {
+							return peersOffersDismissed;
+						}
+					}
+
+					@Override
+					public void set(Boolean val) {
+						synchronized(Node.this) {
+							peersOffersDismissed = val;
+						}
+					}
+				});
+		peersOffersDismissed = nodeConfig.getBoolean("peersOffersDismissed");
+
+		if (peersOffersDismissed) return;
+
+		File[] files = runDir.file("peers-offers").listFiles();
+		if (files != null && files.length > 0) {
+			StringBuilder frefFiles = new StringBuilder();
+			String prefix = "";
+			for (final File file : files) {
+				if (file.isFile()) {
+					String filename = file.getName();
+					if (filename.contains(".") && "fref".equals(filename.substring(filename.lastIndexOf(".") + 1))) {
+						frefFiles.append(prefix).append(file.getName());
+						prefix = ", ";
+					}
+				}
+			}
+			this.clientCore.alerts.register(new PeersOffersUserAlert(frefFiles.toString(), this));
```

Comment:
Also, I would totally move the generation of this `String` into the `UserAlert` and just hand in the relevant files to the `UserAlert`. (Probably using a fake-factory for the alert, i.e. a static `createAlert(node, files)` method.)